uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/annotations/SupportedBrokers.java

```diff
@@ -0,0 +1,40 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.client.ui.automation.annotations;
+
+import com.microsoft.identity.client.ui.automation.broker.ITestBroker;
+
+import java.lang.annotation.Retention;
+
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * An annotation indicating which brokers are supported for a given test. This annotation is coupled
+ * with the {@link com.microsoft.identity.client.ui.automation.rules.BrokerSupportRule} to decide
+ * if a test should be run/skipped.
+ */
+@Retention(value = RUNTIME)
+public @interface SupportedBrokers {
```

Comment:
Do you need Target on here, too?

Reply:
Target sdk?

Reply:
I was wondering, that since it looks like this is only valid for fields and classes, whether we wanted to have the compiler enforce that for us by adding:

@Target(targetType = {ElementType.Field, ElementType.class}) // I think.

Reply:
That syntax may be off, BTW.

Reply:
Yeah I think that makes sense. I'll add that.

Reply:
Addressed here: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1070/commits/4b49a4b913ea063e2b1050e66ac04d8f85606be5

Restricted target to `method` & `type` (there doesn't seem to be one for just `class` -- `type` covers class, interface and enum)